### DISCUSSION TITLE
Set lisp-data-mode for *.el files under /var and /etc directory

### DISCUSSION THREAD

**Discussion by eki3z** [emacscollective/no-littering#164] [Ideas]
Ref: [lisp-data-mode](http://git.savannah.gnu.org/cgit/emacs.git/commit/etc/NEWS?id=43fded12d544ab68f493142debc5b3e437317f35)

`lisp-data-mode` is designed for editing non-code Lisp data, which should be fully applied to the data and history file with lisp format in no-littering. Theoretically, packages should set it by themselves, which has been completed in some builtin packages like `save-place`. However, most famous third-party package like `projectile` don't recognize this.

I think it would be better to set it by no-littering together. After customization of no-littering, some data files has changed names and paths, for example, 

> projectile-cache-file is a variable defined in projectile.el.
> 
> Value
> "/Users/ray/.config/emacs/.cache/var/projectile/cache.el"
> 
> Original Value
> "/Users/ray/.config/emacs/projectile.cache"

maybe add all *.el files under  `no-littering-etc-directory` and `no-littering-var-directory ` to `auto-mode-alist`?
```elisp
(when (fboundp 'lisp-data-mode)
  (add-to-list 'auto-mode-alist
               `(,(concat (regexp-quote no-littering-etc-directory) ".*\\.el\\'")
                 . lisp-data-mode))
  (add-to-list 'auto-mode-alist
               `(,(concat (regexp-quote no-littering-var-directory) ".*\\.el\\'")
                 . lisp-data-mode)))
```

@tarsius How about this?

**Comment by tarsius** [comment]
IMO we should only add this to the [Suggested Settings](https://github.com/emacscollective/no-littering#suggested-settings) section.

**Reply by eki3z** [reply]
Agree.
